lib/db/mysql.js

```diff
@@ -1013,10 +1027,20 @@ module.exports = function (log, error) {
   }
 
   MySql.prototype.getConnection = function (name) {
-    return retryable(
-      this.getClusterConnection.bind(this, name),
-      [ER_TOO_MANY_CONNECTIONS, 'ECONNREFUSED', 'ETIMEDOUT', 'ECONNRESET']
-    )
+    return new P((resolve, reject) => {
+      retryable(
+        this.getClusterConnection.bind(this, name),
+        [ER_TOO_MANY_CONNECTIONS, 'ECONNREFUSED', 'ETIMEDOUT', 'ECONNRESET']
+      ).then((connection) => {
+        connection.query('SET NAMES utf8mb4 COLLATE utf8mb4_bin;', (err) => {
```

Comment:
(We should also copy across the strict-mode enforcement from that repo, but we can do that as a separate PR)